### PR TITLE
Clp editor

### LINKED ISSUE #72: Manage CLPs with Dashboard
Will it eventually be possible to manage Class Level Permissions in the new dashboard like we were able to previously on the hosted dashboard? This would be a very useful feature. Also great job on this, looks great!

### PR DESCRIPTION
**Pull request by drew-gross**
Closes #72

### CODE REVIEW

**Comment by drew-gross** [comment]
Whoops, still need to disable pointer permissions related stuff

File: src/components/PermissionsDialog/PermissionsDialog.react.js [hunk 1]
```diff
@@ -224,22 +224,26 @@ function renderPointerCheckboxes(rowId, publicPerms, pointerPerms, advanced, onC
...
+    enablePointerPermissions = false,
+    permissions,
+    advanced,
+  }) {
     super();
 
     let uniqueKeys = ['*'];
     let perms = {};
-    for (let k in props.permissions) {
+    for (let k in permissions) {
```
Comment (flovilmart) [inline src/components/PermissionsDialog/PermissionsDialog.react.js]: prefer permissions.reduce() if possible

File: src/components/PermissionsDialog/PermissionsDialog.react.js [hunk 2]
```diff
@@ -250,17 +254,17 @@ export default class PermissionsDialog extends React.Component {
     }
 
     let pointerPerms = {};
-    if (props.permissions.readUserFields) {
-      props.permissions.readUserFields.forEach((f) => {
+    if (permissions.readUserFields) {
+      permissions.readUserFields.forEach((f) => {
```
Comment (flovilmart) [inline src/components/PermissionsDialog/PermissionsDialog.react.js]: use reduce

File: src/components/PermissionsDialog/PermissionsDialog.react.js [hunk 3]
```diff
@@ -250,17 +254,17 @@ export default class PermissionsDialog extends React.Component {
...
+        if (permissions.writeUserFields && permissions.writeUserFields.indexOf(f) > -1) {
           p.write = true;
         }
         pointerPerms[f] = Map(p);
       });
     }
-    if (props.permissions.writeUserFields) {
-      props.permissions.writeUserFields.forEach((f) => {
+    if (permissions.writeUserFields) {
+      permissions.writeUserFields.forEach((f) => {
```
Comment (flovilmart) [inline src/components/PermissionsDialog/PermissionsDialog.react.js]: reduce

File: src/dashboard/Data/Browser/Browser.react.js [hunk 4]
```diff
@@ -235,23 +233,12 @@ export default class Browser extends DashboardView {
     });
   }
 
-  fetchCollectionCounts() {
+  handleNewSchema() {
```
Comment (flovilmart) [inline src/dashboard/Data/Browser/Browser.react.js]: maybe rename to handleFetchedSchema (new feels we created something)

File: src/dashboard/Data/Browser/SecurityDialog.react.js [hunk 5]
```diff
@@ -13,8 +13,10 @@ import React             from 'react';
 import styles            from 'dashboard/Data/Browser/Browser.scss';
 
 function validateEntry(pointers, text) {
-  if (pointers.indexOf(text) > -1) {
-    return Parse.Promise.as({ pointer: text });
+  if (false) { //Eventually we will branch on whether or not the server supports pointer permissions
```
Comment (flovilmart) [inline src/dashboard/Data/Browser/SecurityDialog.react.js]: maybe add `const SUPPORTS_POINTER_PERMISSONS=false`

File: src/dashboard/Data/Browser/SecurityDialog.react.js [hunk 6]
```diff
@@ -43,19 +45,13 @@ export default class SecurityDialog extends React.Component {
...
-    });
-  }
-
   render() {
     let dialog = null;
     if (this.props.perms && this.state.open) {
       dialog = (
         <PermissionsDialog
           title='Edit Class Level Permissions'
+          enablePointerPermissions={false /* not supported by Parse Server yet */}
```
Comment (flovilmart) [inline src/dashboard/Data/Browser/SecurityDialog.react.js]: see above

File: src/components/PermissionsDialog/PermissionsDialog.react.js (hunk 1 above)
Comment (drew-gross) [inline src/components/PermissionsDialog/PermissionsDialog.react.js]: I'm actually not a huge fan of using reduce to build up lists/objects. It feels like a lot of boilerplate compared to a for loop. Also this is just a change to use destructuring, I don't really want to go changing other stuff in this function.

Comment (flovilmart) [inline src/components/PermissionsDialog/PermissionsDialog.react.js]: ok!

**Pull request merged**